controllers/sessionController.js

```diff
@@ -0,0 +1,65 @@
+const User = require("../models/User");
+const parseVErr = require("../util/parseValidationErr");
+
+const registerShow = (req, res) => {
+  res.render("register");
+};
+
+const registerDo = async (req, res, next) => {
+  let validation_errors = false
+  if (req.body.password != req.body.password1) {
+    req.flash("error", "The passwords entered do not match.");
+    validation_errors = true
+  }
```

Comment:
Add an else here and add the try-catch block inside the else. Because if you don't do that, even if the password doesn't match, the application will register the user.
```
else {
    try {
      await User.create(req.body);
    } catch (e) {
      if (e.constructor.name === "ValidationError") {
        parseVErr(e, req);
      } else if (e.name === "MongoServerError" && e.code === 11000) {
        req.flash("error", "That email address is already registered.");
      } else {
        return next(e);
      }
      validation_errors = true
    }
  }
```